packages/orm/src/EntityManager.ts

```diff
@@ -180,8 +180,8 @@ export class EntityManager<C = {}> {
   }
 
   /** Looks up `id` in the list of already-loaded entities. */
-  getEntity<T extends Entity>(id: IdOf<T>): T | undefined {
-    return this._entityIndex.get(id) as T | undefined;
+  getEntity<T extends Entity>(tableName: string, id: IdOf<T>): T | undefined {
+    return this._entityIndex.get(`${tableName}:${id}`) as T | undefined;
```

Comment:
Instead of using the `tableName`, I'm thinking we could try something like:

* Add a `BaseEntity.entityId` getter
* If an entity is using tagged ids, `entityId` returns the tagged id as-is
* If an entity is not using tagged ids, `entityId` returns `${constructor.name}:${id}` i.e. `AuthorStat:1`

Then internally in the places we had been using `entity.id` with the assumption it was tagged, we could use `entity.entityId`.

Maybe the only con is that historically we've been cautious about adding too many extra methods onto entity's, b/c typically that is something that purists hate on ORMs about, but :shrug: IMO I could see `entity.entityId` being useful for apps themselves to use, and it seems very unlikely to conflict with columns the user is storing in the database, so should be fine.

Reply:
...a variation on ^ would be that we keep a lot of the tagged id code as-is, so like `AuthorStat` would still get an `as` tag, and `tagName` would go back to being just `string` and not `string | undefined`, but then:

* We allow the user to control whether `someEntity.id` returns the tagged id or the normal id (and that's it)
* We add a `BaseEntity.taggedId` that always returns the tagged id
* Internally we move most callers of `entity.id` that today are assuming "it will always be a tagged id" to use `entity.taggedId`.

I think I like this alternative.... It is similar to the `entity.entityId` idea, of having a way to "always get a unique-per-entity id", but simplifies it by letting that always be a tagged id, even if the user "doesn't want tagged ids" returned from their `.id` method.

It also would mean most of the config / assign tag changes in this PR could go away...

And on the config side we've have something new like:

```
{ id: "tagged" | "untagged" }
```

In the `joist-codegen.json`. And you could either set it at the top-level for all entities, or if you wanted on an entity-by-entity base...

@mmahalwy dunno, do you mind maybe keeping this PR as-is and just doing a brand new branch that spikes this approach?

We could probably do something in `generateEntityCodegenFile` that, when generating the `.id` field, looks at the config `id === tagged / untagged` and output an extra `deTag` to quickly strip the prefix before returning it to the user.

Wdyt?

Reply:
@stephenh that works for me. We can do that.

So to sum it up, the idea is:
1. If you want untagged, it will wrap the id getter with the detag function
2. we will create another getting for getting the tagged id, using it regardless if choosing tagged or untagged

I do think we should have this top level, not per model.

Reply:
Yep! 

> I do think we should have this top level, not per model.

Yeah, that sounds good to me, I agree having it change per-model would be pretty weird.

We might have to find a few more places to unwrap (I'm thinking like `ManyToOneReference.id`?) but `Entity.id` is surely the big one that most users would want to have change.

Reply:
Also if it's only top-level, we'll need a new test project, similar to uuid-ids with ~one-two entities and id: untagged set.

Reply:
feels like we should consider nesting into a `testing-apps` instead of having multiple test projects under `packages`?

Reply:
yep thats what i mean @stephenh